pkg/dockerregistry/server/blobdescriptorservice.go

```diff
@@ -25,6 +38,137 @@ type blobDescriptorService struct {
 	distribution.BlobDescriptorService
 }
 
+// Stat returns a a blob descriptor if the given blob is either linked in repository or is referenced in
+// corresponding image stream. This method is invoked from inside of upstream's linkedBlobStore. It expects
+// a proper repository object to be set on given context by upper openshift middleware wrappers.
 func (bs *blobDescriptorService) Stat(ctx context.Context, dgst digest.Digest) (distribution.Descriptor, error) {
-	return dockerRegistry.BlobStatter().Stat(ctx, dgst)
+	// if there is a repo layer link, return its desc
+	desc, err := bs.BlobDescriptorService.Stat(ctx, dgst)
```

Comment:
> We may need to reorder it rather than turning it off - my worry is that lower code depends on it.

Not really. The order of wrappers and stores is:

    errorBlobStore { pullthroughBlobStore { quotaRestrictedBlobStore { linkedBlobStore { cachedBlobStatter {
       cache: inMemoryBlobDescriptorCacheProvider,
       backend: blobDescriptorService { upstreamBlobDescriptorService },
    }}}}}

The inMemoryCache will be updated with any successful stat we return with our backend and the result will be kept forever. Reordering (moving our blobDescriptorService one level up) is not possible without patching upstream again. I'd rather use `cachedLayers` cache as a replacement for the inMemoryCache because in our case all the layer links and etcd data are ephemeral with proper periodic pruning.